File: Knjigoteka.Model/Entities/OrderItem.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Knjigoteka.Model.Entities
{
    public class OrderItem
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int OrderId { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        [Range(0, int.MaxValue, ErrorMessage = "Quantity must be greater than 0")]
        public int Quantity { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal UnitPrice { get; set; }

        [ForeignKey(nameof(BookId))]
        public Book Book { get; set; } = null!;
        [ForeignKey(nameof(OrderId))]
        public Order Order { get; set; } = null!;
    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/order_item.dart
class OrderItem {
  final int bookId;
  final String title;
  final int quantity;
  final double unitPrice;

  OrderItem({
    required this.bookId,
    required this.title,
    required this.quantity,
    required this.unitPrice,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => OrderItem(
    bookId: json['bookId'],
    title: json['title'],
    quantity: json['quantity'],
    unitPrice: (json['unitPrice'] as num).toDouble(),
  );
}
